web/src/features/modals/AnnouncementModal.tsx

```diff
@@ -0,0 +1,83 @@
+import { SourceColorAnnouncementImage } from 'icons/announcementSourceColorImage';
+import { useAtom } from 'jotai';
+import { atomWithStorage } from 'jotai/utils';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { HiXMark } from 'react-icons/hi2';
+import { hasOnboardingBeenSeenAtom } from 'utils/state/atoms';
+
+const hasAnnouncementBeenSeenAtom = atomWithStorage(
+  'announcementSeen',
+  localStorage.getItem('announcementSeen') ?? false
+);
+
+const handleTierGaterClick = (event: React.MouseEvent<HTMLDivElement>) => {
+  event.stopPropagation();
+  event.preventDefault();
+};
+
+export function AnnouncementModal() {
+  const { t } = useTranslation();
+  const [hasAnnouncementBeenSeen, setHasAnnouncementBeenSeen] = useAtom(
+    hasAnnouncementBeenSeenAtom
+  );
+  const [hasOnboardingBeenSeen] = useAtom(hasOnboardingBeenSeenAtom);
+  const [hasOnboardingBeenSeenInSession, setHasOnboardingBeenSeenInSession] =
+    useState(false);
+  const isExpired = new Date() > new Date('2024-06-30');
+
+  // Check if onboarding has been seen in the current session
+  useEffect(() => {
+    const onboardingSeenInSession = sessionStorage.getItem('onboardingSeenInSession');
+    if (onboardingSeenInSession) {
+      setHasOnboardingBeenSeenInSession(true);
+    } else if (!hasOnboardingBeenSeen) {
+      sessionStorage.setItem('onboardingSeenInSession', 'true');
+      setHasOnboardingBeenSeenInSession(true);
+    }
+  }, [hasOnboardingBeenSeen]);
```

Comment:
Me and @tonypls decided that the announcementModal should only be shown if the user has seen the onboarding modal before but not in the current session, as it we would assume that it would be a new user who would not care that we changed the color. 

I couldn't really figure out a better way of doing this, so suggestions to do it another way a very welcome ⭐

Reply:
Not quite sure I understand why we need both the general check and the session check?

What am I missing with just the logic like the old modal did? :)
```js
const visible =
    Boolean(hasOnboardingBeenSeen) &&
    !hasTotalEnergyIntroBeenSeen &&
    !skipOnboarding &&
    !isExpired;
```

Reply:
The old modal would show up after a user had just seen the onboarding modal, but we decided that a user that has just seen the onboarding modal probably wouldn't be interested in the announcement modal. 

So we only want to show the announcement modal if the user has not seen the onboarding modal in the current session, but it has been seen before in another session.

Reply:
Aaah I see, makes sense then :)